xml/src/main/scala/akka/stream/alpakka/xml/impl/StreamingXmlParser.scala

```diff
@@ -22,16 +22,17 @@ private[xml] object StreamingXmlParser {
 /**
  * INTERNAL API
  */
-@InternalApi private[xml] class StreamingXmlParser(ignoreInvalidChars: Boolean,
-                                                   configureFactory: AsyncXMLInputFactory => Unit)
-    extends GraphStage[FlowShape[ByteString, ParseEvent]] {
-  val in: Inlet[ByteString] = Inlet("XMLParser.in")
-  val out: Outlet[ParseEvent] = Outlet("XMLParser.out")
-  override val shape: FlowShape[ByteString, ParseEvent] = FlowShape(in, out)
+@InternalApi private[xml] class StreamingXmlParser[Ctx](ignoreInvalidChars: Boolean,
```

Comment:
I wonder if we should keep the non-context version around separately to not burden it with unneeded tuple creation. But maybe that would become too much duplication.

Reply:
I suppose we can avoid some duplication by letting callers of the internal API decide how elements in the flow are created, at the expense of complicating the internal API (https://github.com/akka/alpakka/pull/2935/commits/d66f71cb67a5e6e2cbec298dc9e01c6df21f6832). What do you think?